File: javascript/twoSum.js
var twoSum = function(nums, target) {
  const map = {};
  for (let i = 0; i < nums.length; i++) {
    const num = nums[i];
    const diff = target - num;
    if (Number.isInteger(map[diff])) {
      return [map[diff], i];
    }
    map[num] = i;
  }
};

console.log(twoSum([2,7,11,15], 9));

File: python/two_sum.py
class Solution(object):
    def twoSum(self, nums, target):
        seen = {}
        for i, num in enumerate(nums):
            remaining = target - num
            if remaining in seen:
                return [seen[remaining], i]
            seen[num] = i
        return []

sol = Solution()
print(sol.twoSum([], 9))
print(sol.twoSum([1, 2, 3], 5))
print(sol.twoSum([2,7,11,15], 9))
